modules/http/http-worker.c

```diff
@@ -512,7 +515,21 @@ _curl_perform_request(HTTPDestinationWorker *self, HTTPLoadBalancerTarget *targe
 
   curl_easy_setopt(self->curl, CURLOPT_URL, target->url);
   curl_easy_setopt(self->curl, CURLOPT_HTTPHEADER, http_curl_header_list_as_slist(self->request_headers));
-  curl_easy_setopt(self->curl, CURLOPT_POSTFIELDS, self->request_body->str);
+  if (owner->message_compression != CURL_COMPRESSION_UNCOMPRESSED)
+    {
+      if(http_dd_compress_string(self->request_body_compressed, self->request_body, owner->message_compression))
+        {
+          curl_easy_setopt(self->curl, CURLOPT_POSTFIELDS, self->request_body_compressed->str);
+          curl_easy_setopt(self->curl, CURLOPT_POSTFIELDSIZE, self->request_body_compressed->len);
+        }
+      else
+        {
+          curl_easy_setopt(self->curl, CURLOPT_POSTFIELDS, self->request_body->str);
+        }
```

Comment:
[optional]

Should we raise a warning maybe, when we fail to compress?

Reply:
`http_dd_compress_string()` does raise a warning @ line 109 in `http-curl-compress.c`

Reply:
That is true, however we do not notify the user, that we are falling back to non-compression send. I think it would be worth to do that, because from the current message we do not know whether it gets dropped, or tried again or something else.

Reply:
`_handle_compression_error` does write it is sending uncompressed data, as seen in `_compression_error_message' (both in `http-curl-compression.c`), however, this does violate SoC, but that is an other matter.
I will split the warning about the compression error and sending uncompressed data, but only becouse the SoC violation.

Reply:
Done